### DISCUSSION TITLE
Don't consider typo rule on numbers

### DISCUSSION THREAD

**Discussion by macraig** [meilisearch/product#688] [Feedback & Feature Proposal]
Feedback from @JamesToothill  in https://github.com/meilisearch/meilisearch/issues/3999

**Feature description**
When you have typo tolerance enabled with a `oneTypo` setting (in our case of `4`), numeric typos are always considered.  

This means a search for example for:

"2001"

Gets results such as:

"2001: A Space Odyssey"
"2001: A Space Travesty" 
"Dracula 2000"
"Fantasia 2000"

**Current behaviour**
Steps to reproduce the behaviour:
1. Use an index with data that has years in the (searchable) title.
3. Set the `oneTypo` setting to `4` (or less)
4. Search for a year

**Expected behavior**
Algolia has a setting to disable this behaviour `allowTyposOnNumericTokens` which is default `true` but which we would set to `false`.

This allows us to search for:

"2001"

And get results such as:

"2001: A Space Odyssey"
"2001: A Space Travesty" 

But not irrelevant results such as:

"Dracula 2000"
"Fantasia 2000"

**Meilisearch version:**
v1.2.0

**Comment by ManyTheFish** [comment]
Hello @macraig,
A potential implementation could be:
- makes charabia detect the number
- On the search side, removing the allowed typos when generating the Querytree when a word is detected as a number

This would allow us in the future to index numbers differently when tokenizing documents with Charabia during the indexing process

**Reply by thePanz** [reply]
Note: for some searches, a typo on a number is expected and should be allowed.

An example case is searching for ZIP codes

**Reply by macraig** [reply]
A potential broader solution could be to enable/disable typoTolerance as needed: https://github.com/orgs/meilisearch/discussions/727

**Comment by macraig** [comment]
Hello everyone 👋

We have just released the first RC (release candidate) of Meilisearch containing this feature!
You can test it by using

- the [release assets](https://github.com/meilisearch/meilisearch/releases)
- the [Meilisearch Docker image](https://hub.docker.com/repository/docker/getmeili/meilisearch/general)

```
docker run -it --rm -p 7700:7700 -v $(pwd)/meili_data:/meili_data getmeili/meilisearch:v1.15.0-rc.0

```

You are welcome to leave your feedback in this discussion.

If you encounter any bugs, please report them [here](https://github.com/meilisearch/meilisearch/issues/new?assignees=&labels=&template=bug_report.md&title=).
Thanks in advance for your help and your involvement in Meilisearch ❤️

🎉 Official and stable release containing this change will be available on `June 9th`, `2025`

⚠️ RC (release candidates) are not recommended for production

**Comment by macraig** [comment]
Hey folks 👋 

[v1.15](https://github.com/meilisearch/meilisearch/releases/tag/v1.15.0) has been released! 🦘   You can now disable type tolerance on numeric fields ✨

**Reply by JamesToothill** [reply]
My hero 🙌
